pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -0,0 +1,337 @@
+		needDrain = true
+		needReboot = true
+		return
+	}
+	if p.updateTarget.needUpdate() {
+		glog.Infof("k8s-plugin OnNodeStateChange(): needDrain to update %q", p.updateTarget)
+		needDrain = true
+	}
+
+	return
+}
+
+// Apply config change
+func (p *K8sPlugin) Apply() error {
+	glog.Info("k8s-plugin Apply()")
+	if err := p.updateSwichdevService(); err != nil {
+		return err
+	}
+
+	for _, systemService := range p.updateTarget.systemServices {
+		if err := p.updateSystemService(systemService); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *K8sPlugin) readSwitchdevManifestFiles() error {
+	if p.switchdevService == nil {
+		switchdevService, err := service.ReadServiceManifestFile(switchdevUnitFile)
+		if err != nil {
+			return err
+		}
+
+		// Remove run condition form the service
+		conditionOpt := &unit.UnitOption{
+			Section: "Unit",
+			Name:    "ConditionPathExists",
+			Value:   "!/etc/ignition-machine-config-encapsulated.json",
+		}
+		switchdevService, err = service.RemoveFromService(switchdevService, conditionOpt)
+		if err != nil {
+			return err
+		}
+		p.switchdevService = switchdevService
+	}
+
+	if p.switchdevRunScript == nil {
+		switchdevRunScript, err := service.ReadScriptManifestFile(configuresSwitchdevScript)
+		if err != nil {
+			return err
+		}
+
+		p.switchdevRunScript = switchdevRunScript
+	}
+
+	return nil
+}
+
+func (p *K8sPlugin) readManifestFiles() error {
+	err := p.readSwitchdevManifestFiles()
+	if err != nil {
+		return err
+	}
+
+	if p.networkManagerService == nil {
+		p.networkManagerService, err = service.ReadServiceInjectionManifestFile(networkManagerUnitFile)
+		if err != nil {
+			return err
+		}
+	}
+
+	if p.openVSwitchService == nil {
+		p.openVSwitchService, err = service.ReadServiceInjectionManifestFile(ovsUnitFile)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *K8sPlugin) switchdevServiceStateUpdate() error {
+	// Check switchdev service
+	swdService, err := p.serviceManager.ReadService(p.switchdevService.Path)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		// service not exists
+		p.updateTarget.switchdevService = true
+	} else {
+		needChange, err := service.CompareServices(swdService, p.switchdevService)
+		if err != nil {
+			return err
+		}
+		p.updateTarget.switchdevService = needChange
+	}
+
+	// Check switchdev run script
+	data, err := ioutil.ReadFile(path.Join(chroot, p.switchdevRunScript.Path))
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		p.updateTarget.switchdevScript = true
+	} else if string(data) != p.switchdevRunScript.Contents.Inline {
+		p.updateTarget.switchdevScript = true
+	}
+
+	return nil
+}
+
+func (p *K8sPlugin) getSystemServices() []*service.Service {
+	return []*service.Service{p.networkManagerService, p.openVSwitchService}
+}
+
+func (p *K8sPlugin) isSystemServiceNeedUpdate(serviceObj *service.Service) (bool, error) {
```

Comment:
let split the logic. 
You should have isSystemServiceExist() to validate at start 
and then isSystemServiceNeedUpdate can return just bool